usbsdmux/service.py

```diff
@@ -0,0 +1,144 @@
+#! /usr/bin/env python3
+
+from .usbsdmux import UsbSdMux
+import socket
+import time
+import argparse
+import json
+import os
+
+"""
+This service is intended as systemd-socket-activated unit and provides an
+interface to the USB-SD-Mux without the need for root privileges.
+
+The service uses a SOCK_SEQPACKET UNIX-socket for communication with the a
+client. Packets always contain a JSON-encoded dict as payload.
+
+From client to service:
+The service expects the following parameter inside the payload:
+* 'sg'   -- /dev/sg* - device to use to control the USB-SD-Mux
+* 'mode' -- Mode to set the USB-SD-Mux to
+
+For example:
+{"sg": "/dev/sg1", "mode": "dut"}
+
+From service to client:
+The service always answers with a payload like the following:
+* 'error' -- Boolean: True if an error occurred. False on success.
+* 'text'  -- Text that describes the result.
+             Contains the Text of an exception if one occurred.
+
+For example:
+{"error": False, "text": "Success"}
+"""
+# Default filedescriptor used by systemd to pass us a socket
+systemd_socket_fd = 3
+
+def answer(had_error=False, err_text=""):
+    """
+    Creates a JSON-formatted answer for socket-communication.
+
+    Arguments:
+    had_error -- Defines if the return value represents an error. Will be
+                 written into the 'error'-field of the answer.
+    err_text -- Free Text that will be passed to the client along with the
+                error state.
+    """
+    answer = dict()
+    answer["error"] = had_error
+    answer["text"] =  err_text
+    return json.dumps(answer)
+
+def parse(raw_string):
```

Comment:
This executes the command as well as parse it. Maybe rename it to something else?

Reply:
You are right. The name is misleading.

Renamed:
* :code:`parse()` to :code:`process_request()`
* :code:`answer()` to :code:`create_answer()`